src/pages/iou/request/step/withWritableReportOrNotFound.tsx

```diff
@@ -54,16 +51,23 @@ export default function <TProps extends WithWritableReportOrNotFoundProps<MoneyR
     shouldIncludeDeprecatedIOUType = false,
 ): React.ComponentType<Omit<TProps & RefAttributes<TRef>, keyof WithWritableReportOrNotFoundOnyxProps>> {
     // eslint-disable-next-line rulesdir/no-negated-variables
-    function WithWritableReportOrNotFound(props: TProps, ref: ForwardedRef<TRef>) {
-        const {report = {reportID: ''}, route, isLoadingApp = true, reportDraft} = props;
+    function WithWritableReportOrNotFound(props: Omit<TProps, keyof WithWritableReportOrNotFoundOnyxProps>, ref: ForwardedRef<TRef>) {
```

Comment:
We must remove OnyxProps https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#hooks-and-hocs

Reply:
import type {RouteProp} from '@react-navigation/core';
import type {ComponentType, ForwardedRef, RefAttributes} from 'react';
import React, {forwardRef, useEffect} from 'react';
import {useOnyx} from 'react-native-onyx';
import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
import getComponentDisplayName from '@libs/getComponentDisplayName';
import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
import * as ReportUtils from '@libs/ReportUtils';
import * as ReportActions from '@userActions/Report';
import CONST from '@src/CONST';
import ONYXKEYS from '@src/ONYXKEYS';
import type SCREENS from '@src/SCREENS';

type MoneyRequestRouteName =
    | typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT
    | typeof SCREENS.MONEY_REQUEST.STEP_DESCRIPTION
    | typeof SCREENS.MONEY_REQUEST.STEP_DATE
    | typeof SCREENS.MONEY_REQUEST.STEP_CATEGORY
    | typeof SCREENS.MONEY_REQUEST.STEP_DISTANCE_RATE
    | typeof SCREENS.MONEY_REQUEST.STEP_CONFIRMATION
    | typeof SCREENS.MONEY_REQUEST.STEP_TAX_RATE
    | typeof SCREENS.MONEY_REQUEST.STEP_AMOUNT
    | typeof SCREENS.MONEY_REQUEST.STEP_DISTANCE
    | typeof SCREENS.MONEY_REQUEST.CREATE
    | typeof SCREENS.MONEY_REQUEST.START
    | typeof SCREENS.MONEY_REQUEST.STEP_TAG
    | typeof SCREENS.MONEY_REQUEST.STEP_PARTICIPANTS
    | typeof SCREENS.MONEY_REQUEST.STEP_MERCHANT
    | typeof SCREENS.MONEY_REQUEST.STEP_TAX_AMOUNT
    | typeof SCREENS.MONEY_REQUEST.STEP_SCAN
    | typeof SCREENS.MONEY_REQUEST.STEP_SEND_FROM
    | typeof SCREENS.MONEY_REQUEST.STEP_COMPANY_INFO;

type Route<T extends MoneyRequestRouteName> = RouteProp<MoneyRequestNavigatorParamList, T>;

type WithWritableReportOrNotFoundProps<T extends MoneyRequestRouteName> = {
    route: Route<T>;
};

export default function <TProps extends WithWritableReportOrNotFoundProps<MoneyRequestRouteName>, TRef>(
    WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
    shouldIncludeDeprecatedIOUType = false,
): React.ComponentType<TProps & RefAttributes<TRef>> {
    // eslint-disable-next-line rulesdir/no-negated-variables
    function WithWritableReportOrNotFound(props: TProps, ref: ForwardedRef<TRef>) {
        const {route} = props;
        const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${route.params.reportID ?? '-1'}`);
        const [isLoadingApp = true] = useOnyx(ONYXKEYS.IS_LOADING_APP);
        const [reportDraft] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_DRAFT}${route.params.reportID ?? '-1'}`);
...
    WithWritableReportOrNotFound.displayName = `withWritableReportOrNotFound(${getComponentDisplayName(WrappedComponent)})`;

    return forwardRef(WithWritableReportOrNotFound);
}

export type {WithWritableReportOrNotFoundProps};

Reply:
> ```
> import type {RouteProp} from '@react-navigation/core';
> import type {ComponentType, ForwardedRef, RefAttributes} from 'react';
> import React, {forwardRef, useEffect} from 'react';
> import {useOnyx} from 'react-native-onyx';
> import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
> import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
> import getComponentDisplayName from '@libs/getComponentDisplayName';
> import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
> import * as ReportUtils from '@libs/ReportUtils';
> import * as ReportActions from '@userActions/Report';
> import CONST from '@src/CONST';
> import ONYXKEYS from '@src/ONYXKEYS';
> import type SCREENS from '@src/SCREENS';
> 
> type MoneyRequestRouteName =
>     | typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT
>     | typeof SCREENS.MONEY_REQUEST.STEP_DESCRIPTION
>     | typeof SCREENS.MONEY_REQUEST.STEP_DATE
>     | typeof SCREENS.MONEY_REQUEST.STEP_CATEGORY
>     | typeof SCREENS.MONEY_REQUEST.STEP_DISTANCE_RATE
>     | typeof SCREENS.MONEY_REQUEST.STEP_CONFIRMATION
>     | typeof SCREENS.MONEY_REQUEST.STEP_TAX_RATE
>     | typeof SCREENS.MONEY_REQUEST.STEP_AMOUNT
>     | typeof SCREENS.MONEY_REQUEST.STEP_DISTANCE
>     | typeof SCREENS.MONEY_REQUEST.CREATE
>     | typeof SCREENS.MONEY_REQUEST.START
>     | typeof SCREENS.MONEY_REQUEST.STEP_TAG
>     | typeof SCREENS.MONEY_REQUEST.STEP_PARTICIPANTS
>     | typeof SCREENS.MONEY_REQUEST.STEP_MERCHANT
>     | typeof SCREENS.MONEY_REQUEST.STEP_TAX_AMOUNT
>     | typeof SCREENS.MONEY_REQUEST.STEP_SCAN
>     | typeof SCREENS.MONEY_REQUEST.STEP_SEND_FROM
>     | typeof SCREENS.MONEY_REQUEST.STEP_COMPANY_INFO;
> 
> type Route<T extends MoneyRequestRouteName> = RouteProp<MoneyRequestNavigatorParamList, T>;
> 
> type WithWritableReportOrNotFoundProps<T extends MoneyRequestRouteName> = {
>     route: Route<T>;
> };
> 
> export default function <TProps extends WithWritableReportOrNotFoundProps<MoneyRequestRouteName>, TRef>(
>     WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
>     shouldIncludeDeprecatedIOUType = false,
> ): React.ComponentType<TProps & RefAttributes<TRef>> {
>     // eslint-disable-next-line rulesdir/no-negated-variables
>     function WithWritableReportOrNotFound(props: TProps, ref: ForwardedRef<TRef>) {
>         const {route} = props;
>         const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${route.params.reportID ?? '-1'}`);
>         const [isLoadingApp = true] = useOnyx(ONYXKEYS.IS_LOADING_APP);
>         const [reportDraft] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_DRAFT}${route.params.reportID ?? '-1'}`);
> ...
>     WithWritableReportOrNotFound.displayName = `withWritableReportOrNotFound(${getComponentDisplayName(WrappedComponent)})`;
> 
>     return forwardRef(WithWritableReportOrNotFound);
> }
> 
> export type {WithWritableReportOrNotFoundProps};
> ```

Just an idea, we have to test it properly